### DISCUSSION TITLE
是否可以保留 NET Framework 上 经典开发思维

### DISCUSSION THREAD

**Discussion by JHeLiu** [dotnet/aspnetcore#23838] [General]
1、比如我们习惯了 C#给我们的帮助文档  using 就代表彻底释放 非托管资源,典型的 HttpClient 实现了释放,却没真正释放
2、ASP NET CORE 太多几乎都是异步，如果设计可以类似于之前的同步不可以吗?
3、在.NET CORE 上太多依赖 DI , 也许这是未来的趋势，或者是说超大型项目更好的解决方式,微软没考虑过很多是中小型项目？
4、太多项目可以不使用DI，原因很多项目是几个人甚至一个人完成的。几年甚至更长的时间不会迁移数据库跟服务层
5、.NET CORE 可以新增很多新功能，但是为什么不保留 几十年来 经典的开发思维,很疑惑?
6、应该大多数开发人员 在写项目的时候会 封装一些属于自己的 扩展类库供团队开发或者个人使用，他们会封装的非常傻瓜式，也许调用者可能只要知道怎么 new 就行了,DI 显然会带来额外的工作量，看似有用 好像又没多大用处
7、希望.NET CORE 可以保留 经典的开发模式 跟 新增官方自带的 DI 开发模式， 这样开发者可以根据不同的业务来选型不同更适合自己业务的技术型
不知道这些有没有安排在 路上

英文不好，只能用中文了， 有没有那个大哥帮忙翻译一下

**Comment by fawdlstty** [comment]
1、用于连接池的情况，HttpClient本身可以用于链接复用，以前方式每个外部请求都得分配、释放，使得频繁外部请求的情况下资源利用率非常低
2、asp.net core并没限制你用同步还是异步，两者交替、只同步、只异步都行
3、DI本身非常适合中小型项目，据我所知只有DDD模型不适合小项目
4、DI只是一种开发方式，跟三层架构差不多都是一种习惯，习惯了就行了
5、你说的那个不叫保留几十年前的开发方式，这个叫历史包袱。.NET Core是一种新框架，不存在历史包袱，所以能飞得更快
6、DI是目前已知开发量最小，能最便捷调用其他库的方式，说白了就是你对DI不太熟
7、没必要，DI已经足够优秀了，足够适合任意类型任意体积项目，以前的开发方式非常垃圾，没必要保留

**Reply by JHeLiu** [reply]
恩，你说的也没错，其实异步跟同步还有DI 这个很容易接受， 现在我们遇到是 HttpClient 这个类被 如果使用官方手段必须要使用DI， 不知道官方有没有可以不使用DI完美使用， 静态使用依然有DNS刷新问题存在。
我们遇到就是封装了一成HTTP请求的核心库做为 内部业务 NUGET包发布使用，现在不的不用 函数公开 HttpClient  服务注入进去。 但是又不想注入跟公开传入 HttpClient 这个类型， 像平常NEW 或者静态调用来的方便。调用的人 根本不关心 甚至不想传入这个“无用”的 类型

**Reply by fawdlstty** [reply]
不使用DI框架，比如控制台情况，调用HttpClient，也行，你需要的是多去百度一下，比如 <https://blog.csdn.net/catshitone/article/details/87922650>

**Comment by fawdlstty** [comment]
This is a boring topic that native English speakers don't need to look at

**Comment by AlphaYu** [comment]
为什么每次搜索微服务，到显示这条信息。。看起来没啥关系啊。
